scene/3d/ray_cast.cpp

```diff
@@ -143,8 +143,12 @@ void RayCast::_notification(int p_what) {
 
 			bool prev_collision_state = collided;
 			_update_raycast_state();
-			if (prev_collision_state != collided && get_tree()->is_debugging_collisions_hint())
-				_update_debug_shape();
+			if (prev_collision_state != collided && get_tree()->is_debugging_collisions_hint()) {
+				if (debug_material.is_valid()) {
+					Ref<FixedSpatialMaterial> line_material = static_cast<Ref<FixedSpatialMaterial> >(debug_material);
```

Comment:
We could if we changed the debug_material to be a FixedSpatialMaterial, but then we would have to add an include to the header or forward-declare FixedSpatialMaterial in the header (not sure if that would work in this case). I prefer the cast and we only do it here and in `_create_debug_shape` so it's not too cumbersome in my opinion.